### PR TITLE
[NEURON][PT][BUILD][TRN1][EC2, ECS] - Release PT1.13 inference with neuronx SDK 2.9.0

### PR DESCRIPTION
**Pull request by catalin-manciu-aws**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aws-ceenugal — APPROVED** [review]

**Comment by tejaschumbalkar** [comment]
Please update the PR description

File: pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    cython \
+    mkl-include \
+    mkl \
+    parso \
+    scipy \
+    typing \
+    # Below 2 are included in miniconda base, but not mamba so need to install
+    conda-content-trust \
+    charset-normalizer \
+# Upstream conda looks to have moved to 4.13 which is incompatible with mamba 0.22.1 and will fail the conda-forge installs.
```
Comment (tejaschumbalkar) [inline pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx:84]: is this still applicable?

File: pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+RUN conda install -c conda-forge \
+    opencv \
+    scikit-learn \
+    h5py \
+    requests \
+ && conda clean -ya \
+ && pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3 \
+ && pip install packaging==20.4 \
```
Comment (tejaschumbalkar) [inline pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx:99]: why are we pinning it. Please go through the entire dockerfile and unpin the versions if they are not required to pin explicitly. Add reasoning to the pin packages.

File: test/sagemaker_tests/pytorch/inference/integration/__init__.py [hunk 3]
```diff
@@ -39,6 +40,10 @@
 resnet_neuron_script = os.path.join(model_neuron_dir, code_sub_dir, 'resnet18.py')
 resnet_neuron_input = os.path.join(model_neuron_dir, 'cat.jpg')
 resnet_neuron_image_list = os.path.join(model_neuron_dir, 'imagenet1000_clsidx_to_labels.txt')
+model_neuronx_dir = os.path.join(resnet_path, neuronx_sub_dir)
```
Comment (tejaschumbalkar) [inline test/sagemaker_tests/pytorch/inference/integration/__init__.py:43]: are these file duplicates? if so, host it in a common folder name and refer the path here. We do not want duplicate files

File: test/dlc_tests/eks/pytorch/inference/test_eks_pytorch_inference.py [hunk 4]
```diff
@@ -49,13 +47,19 @@ def test_eks_pytorch_neuron_inference(pytorch_inference_neuron):
...
         assert test_utils.request_pytorch_inference_densenet(
-            port=port_to_forward, server_type=server_type, model_name="pytorch-resnet-neuron"
+            port=port_to_forward, server_type=server_type, model_name=model_name
         )
     finally:
         run(f"kubectl delete deployment {selector_name}")
         run(f"kubectl delete service {selector_name}")
 
 
+@pytest.mark.model("resnet")
```
Comment (tejaschumbalkar) [inline test/dlc_tests/eks/pytorch/inference/test_eks_pytorch_inference.py:57]: is this supposed to be neuron or neuronx? If neuron, lets create a separate PR for this as this PR concerns neuronx only

Comment (catalin-manciu-aws) [reply test/dlc_tests/eks/pytorch/inference/test_eks_pytorch_inference.py:57]: I had to restructure the code here to allow for neuronx without copying any code. The initial test_eks_pytorch_neuron_inference function below has been broken into: a common part  __run_pytorch_neuron_inference and two tests (the neuronx one has just been added in the latest commit).

File: test/sagemaker_tests/pytorch/inference/integration/__init__.py (hunk 3 above)
Comment (catalin-manciu-aws) [reply test/sagemaker_tests/pytorch/inference/integration/__init__.py:43]: Removed the image and the labels.txt file, the other two files are unique to neuronx

File: pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx (hunk 1 above)
Comment (catalin-manciu-aws) [reply pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx]: Removed

File: pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx (hunk 2 above)
Comment (catalin-manciu-aws) [reply pytorch/inference/docker/1.13/py3/sdk2.9.0/Dockerfile.neuronx]: Reduced pinning as much as possible

**Review by tejaschumbalkar — APPROVED** [review]

**Review by aws-ceenugal — APPROVED** [review]

**Pull request merged**